bitcoin/src/amount.rs

```diff
@@ -517,6 +517,16 @@ impl Amount {
     #[deprecated(since = "0.31.0", note = "Use Self::MIN instead")]
     pub const fn min_value() -> Amount { Amount(u64::min_value()) }
 
+    /// Creates an [`Amount`] from a value representing a whole number of bitcoin.
+    ///
+    /// This function handles an input arg that would overflow the inner type in the following ways:
+    ///
+    /// * In release mode non-const context: silently overflows
+    /// * In release mode const context: silently overflows
```

Comment:
I think there is more nuance than that.  const context will fail to build in release mode also on stable:
 https://play.rust-lang.org/?version=stable&mode=release&edition=2021&gist=b9a25406aad51d58286e2b8ed0603ef3

I'm not sure why you say it silently overflows in const context release mode.